tests/Unit/Domain/DomainCreators/Test_Shell.cpp

```diff
@@ -417,3 +429,111 @@ SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Shell.Factory.ShellWedges",
         which_wedges);
   }
 }
+
+namespace {
+//Tests that the underlying element structure is the same regardless of
+//whether they are created through InitialRefinement or RadialBlockLayers.
+//This is done by constructing a set of expected boundary points dependent on
+//only the total number of Elements in the Domain. The Domain is then created
+//using either InitialRefinement or RadialBlockLayers. Comparing the resulting
+//Element boundary points against the expected points ensures that both methods
+//of creating Elements lead to the same boundary locations.
+//To verify that the Block structures created are indeed distinct, this test
+//also checks the Block structure.
+void test_radial_block_layers(const double inner_radius,
+                              const double outer_radius,
+                              const size_t refinement_level,
+                              const bool use_logarithmic_map,
+                              const size_t radial_block_layers,
+                              const std::vector<size_t>& expected_block_ids) {
+  const size_t number_of_divisions =
+      radial_block_layers * two_to_the(refinement_level);
+  const std::array<size_t, 2> grid_points_r_angular{{2, 3}};
+  const bool use_equiangular_map = true;
+  const double aspect_ratio = 1.0;
+  const ShellWedges which_wedges = ShellWedges::OneAlongMinusX;
+  // Points on the interior of Blocks used to check Block structure:
+  DataVector x_interior(number_of_divisions + 1);
+  // Points on the boundary of Elements used to check Element structure:
+  DataVector x_boundary(number_of_divisions + 1);
+  for (size_t i = 0; i < number_of_divisions + 1; i++) {
+    const double delta =
+        static_cast<double>(i) / static_cast<double>(number_of_divisions);
+    if (use_logarithmic_map) {
+      x_interior[i] = inner_radius * pow(outer_radius / inner_radius, delta);
+      x_boundary[i] = (inner_radius - 0.1) *
+                      pow((outer_radius + 0.1) / (inner_radius - 0.1), delta);
+    } else {
+      x_interior[i] = inner_radius + delta * (outer_radius - inner_radius);
+      x_boundary[i] = (inner_radius - 0.1) +
+                      delta * ((outer_radius + 0.1) - (inner_radius - 0.1));
+    }
+  }
+  const auto zero = make_with_value<DataVector>(x_interior, 0.0);
+  tnsr::I<DataVector, 3, Frame::Inertial> interior_inertial_coords{
+      {{-x_interior, zero, zero}}};
+  const DomainCreators::Shell<Frame::Inertial> shell{
+      inner_radius - 0.1,    outer_radius + 0.1,  refinement_level,
```

Comment:
Why is the inner radius `inner_radius - 0.1`?

Reply:
`block_logical_coordinates` will fail if one of the inertial coords lies outside the domain. To ensure this doesn't happen, I make the domain slightly larger than the range of inertial coords to be tested. Because I create the same number of interior points as boundary points, one block will correspond to two inertial points. Which block this is depends on whether the points are distributed linearly or exponentially. In the linear case this happens in the middle, and in the exponential case this happens at the end. This lists at the end are used to verify that the correct block structure is obtained. I hope this clarification addresses your other comments as well.

Reply:
You only call `block_logical_coordinates` on the interior points, so you shouldn't be putting any of them on the boundary.  Looking at your chosen coordinates above, it looks like one of your "interior" points (the center one) is not in the interior of an element in the linear case.  Just arrange for your interior points to actually be in the interior of the elements and you won't need any of these adjustments.

The easiest way to do this is to put them between the boundary points.  There are various ways to accomplish that.